deploy/TaskTreasuryUpgradable.deploy.js

```diff
@@ -0,0 +1,43 @@
+const { sleep } = require("@gelatonetwork/core");
+const { getGelatoAddress } = require("../hardhat/config/addresses");
+
+module.exports = async (hre) => {
+  const TASKTREASURY = "0x66e2f69df68c8f56837142be2e8c290efe76da9f";
+
+  if (hre.network.name !== "hardhat") {
+    console.log(
+      `Deploying TaskTreasuryUpgradable to ${hre.network.name}. Hit ctrl + c to abort`
+    );
+    console.log(`Using Old TaskTreasury: ${TASKTREASURY}`);
+    await sleep(10000);
+  }
+
+  const { deployments } = hre;
+  const { deploy } = deployments;
+  const { deployer } = await hre.getNamedAccounts();
+  const GELATO = getGelatoAddress(hre.network.name);
+
+  await deploy("TaskTreasuryUpgradable", {
+    from: deployer,
+    proxy: {
```

Comment:
to update state on proxy, you need to pass the args in the init function like so: 
```
 proxy: {
      proxyContract: "EIP173ProxyWithReceive",
      owner: deployer,
      execute: {
        init: {
          methodName: "initialize",
          args: [GELATO, TASKTREASURY],
        },
      },
    },
```